### DISCUSSION TITLE
MinIO upgrade

### DISCUSSION THREAD

**Discussion by sumarnaz** [minio/minio#13571] [Q&A]
Hello.
I have a question concerning the MinIO upgrade.
I'm currently running a 4-nodes cluster with 2 MinIO instances (one on port 9000 and the other on port 9001).
The running version is  Version: 2021-06-09T18:51:39Z on both instances (since the minio executable is the same)

```
[~] mc admin info hostit-backup
●  <minio-server1>:9000
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 

●  <minio-server2>:9000
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 

●  <minio-server3>:9000
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 

● <minio-server4>:9000
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 
```
```
[~] mc admin info public-buckets
●  <minio-server1>:9001
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 

●  <minio-server2>:9001
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 

● <minio-server3>:9001
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 

●  <minio-server4>:9001
   Uptime: 4 months 
   Version: 2021-06-09T18:51:39Z
   Network: 4/4 OK 
   Drives: 12/12 OK 
```
In order to upgrade it, as far as I understand, I should run the following command:

`mc admin update [alias]`

The questions are: 

1) should I execute the command just once 

`[~] mc admin update hostit-backup`

or should I execute both?
```
[~] mc admin update hostit-backup
[~] mc admin update public-buckets
```
2) should I stop the instances before running the upgrade command?

Thank you very much

Riccardo

**Comment by klauspost** [comment]
1. Both since they don't know about eachother.
2. No. Then the server cannot be reached.

**Reply by sumarnaz** [reply]
Hello.
Thank you very much for the quick reply!

1. I was confused because they actually use the same minio executable, so I though that on the first command the new version would have been available for both.
2. Do you think it would be anyway better not to have active writes/reads during the upgrade procedure?

Kind regards
Riccardo

**Reply by harshavardhana** [reply]
If they use both on the same server you have to make sure to replace the binary manually and restart the service.

**Reply by sumarnaz** [reply]
Hello.
... so I'm confused: there is no need to execute mc admin update [alias]?  
Just:

```
wget https://dl.min.io/server/minio/release/darwin-amd64/minio
chmod +x minio
..
systemctl start
```

Cheers
Riccardo

**Reply by harshavardhana** [reply]
> wget https://dl.min.io/server/minio/release/darwin-amd64/minio
> chmod +x minio
> ..
> systemctl start

You are not Linux make sure to download Linux binary.

**Reply by sumarnaz** [reply]
Hello.
Yes, sorry... I copy-pasted the wrong lines..

```
wget https://dl.min.io/server/minio/release/linux-amd64/minio
chmod +x minio
..
```

Thank you very much
Riccardo

**Comment by harshavardhana** [comment]
> Hello. I have a question concerning the MinIO upgrade. I'm currently running a 4-nodes cluster with 2 MinIO instances (one on port 9000 and the other on port 9001). The running version is Version: 2021-06-09T18:51:39Z on both instances (since the minio executable is the same)
> 
> ```
> [~] mc admin info hostit-backup
> ●  <minio-server1>:9000
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> 
> ●  <minio-server2>:9000
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> 
> ●  <minio-server3>:9000
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> 
> ● <minio-server4>:9000
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> ```
> 
> ```
> [~] mc admin info public-buckets
> ●  <minio-server1>:9001
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> 
> ●  <minio-server2>:9001
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> 
> ● <minio-server3>:9001
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> 
> ●  <minio-server4>:9001
>    Uptime: 4 months 
>    Version: 2021-06-09T18:51:39Z
>    Network: 4/4 OK 
>    Drives: 12/12 OK 
> ```
> 

FYI @sumarnaz you may be violating AGPLv3 if you are using MinIO to build commercial products or using it for commercial purposes you are advised to speak to a software legal entity to get an understanding if you are not under violation.

Please use https://min.io/pricing and https://opensource.org/licenses/AGPL-3.0 to understand how you can achieve compliance.

**Reply by Crest** [reply]
Are you trying to imply the AGPLv3 doesn’t allow hosting services for third parties with unmodified builds?!?!?
